src/inner-slider.js

```diff
@@ -387,16 +387,19 @@ export class InnerSlider extends React.Component {
     this.setState(state, () => {
       asNavFor && asNavFor.innerSlider.slideHandler(index);
       if (!nextState) return;
-      this.animationEndCallback = setTimeout(() => {
-        const { animating, ...firstBatch } = nextState;
-        this.setState(firstBatch, () => {
-          this.callbackTimers.push(
-            setTimeout(() => this.setState({ animating }), 10)
-          );
-          afterChange && afterChange(state.currentSlide);
-          delete this.animationEndCallback;
-        });
-      }, speed);
+      this.animationEndCallback = setTimeout(
+        () => {
+          const { animating, ...firstBatch } = nextState;
+          this.setState(firstBatch, () => {
+            this.callbackTimers.push(
+              setTimeout(() => this.setState({ animating }), 10)
+            );
+            afterChange && afterChange(state.currentSlide);
+            delete this.animationEndCallback;
+          });
+        },
+        dontAnimate ? 0 : speed
+      );
```

Comment:
Lines 390Formated by formatter, line 401 is the main change. 

Calling slick's `slickGoTo` more than once has the effect that calls that complete after the first call's setTimeout's function execution to never call their own afterChange (line 387 above), a race condition 🏃‍♂️.